### PR TITLE
[AIRFLOW-5498] Move GCP Discovery hook to core

### PR DESCRIPTION
**Pull request by turbaszek**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5498

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by turbaszek** [comment]
This PR will require note after #6119

File: airflow/contrib/hooks/google_discovery_api_hook.py [hunk 1]
```diff
@@ -16,132 +16,14 @@
...
-        api_responses = []
-
-        while google_api_endpoint_instance:
-            api_response = google_api_endpoint_instance.execute(num_retries=num_retries)
-            api_responses.append(api_response)
-
-            google_api_endpoint_instance = self._build_next_api_request(
-                google_api_conn_client, api_endpoint_parts[1:], google_api_endpoint_instance, api_response
-            )
+"""This module is deprecated. Please use `airflow.gcp.hooks.discovery_api`."""
```
Comment (feluelle) [inline airflow/contrib/hooks/google_discovery_api_hook.py]: Actually, this Hook hasn't been released yet. So I think we can just move it to core without a note.

Comment (feluelle) [reply airflow/contrib/hooks/google_discovery_api_hook.py]: Same for the `GoogleApiToS3Transfer`

Comment (mik-laj) [reply airflow/contrib/hooks/google_discovery_api_hook.py]: I agree. Can you also look at other operators and hooks?

Comment (turbaszek) [reply airflow/contrib/hooks/google_discovery_api_hook.py]: Good point

**Review by mik-laj — APPROVED** [review]

**Pull request merged**